### PR TITLE
implement UserMapper, UserController added model UserResponseDto

### PR DESCRIPTION
**Pull request by IwalkInSilence**

### CODE REVIEW

File: src/main/java/spring/intro/config/WebConfig.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package spring.intro.config;
+
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.web.servlet.config.annotation.EnableWebMvc;
+
+@Configuration
+@EnableWebMvc
+@ComponentScan(basePackages = {
+        "spring.intro.controllers"
```
Comment (semyonich) [inline src/main/java/spring/intro/config/WebConfig.java]: use singular names for packages

File: src/main/java/spring/intro/dao/impl/UserDaoImpl.java [hunk 2]
```diff
@@ -45,4 +45,13 @@ public void add(User user) {
...
         }
     }
+
+    @Override
+    public User get(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            return session.get(User.class, id);
+        } catch (Exception e) {
+            throw new RuntimeException("Error retrieving all users", e);
+        }
```
Comment (semyonich) [inline src/main/java/spring/intro/dao/impl/UserDaoImpl.java:55]: you have only one user here )

**Review by semyonich — CHANGES_REQUESTED** [review]

**Review by semyonich — APPROVED** [review]

File: src/main/java/spring/intro/controller/UserController.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    }
+
+    @GetMapping("/{id}")
+    UserResponseDto get(@PathVariable Long id) {
+        User user = userService.get(id);
+        return userMapper.parseToDto(user);
+    }
+
+    @GetMapping("/")
+    List<UserResponseDto> getall() {
```
Comment (dmytro-hryhoruk) [inline src/main/java/spring/intro/controller/UserController.java]:
```suggestion
    List<UserResponseDto> getAll() {
```

File: src/main/java/spring/intro/dto/UserResponseDto.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
+package spring.intro.dto;
+
+public class UserResponseDto {
+    private String name;
+    private String email;
```
Comment (dmytro-hryhoruk) [inline src/main/java/spring/intro/dto/UserResponseDto.java:6]: let's add id too

**Review by dmytro-hryhoruk — CHANGES_REQUESTED** [review]

File: src/main/java/spring/intro/controller/UserController.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+        User bruce = new User();
+        bruce.setName("Bruce");
+        bruce.setLastname("Willis");
+        bruce.setEmail("alice@lolosh.com");
+        userService.add(bruce);
+        return "All users injected";
+    }
+
+    @GetMapping("/{id}")
+    UserResponseDto get(@PathVariable Long id) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/spring/intro/controller/UserController.java:51]: access modifier

File: src/main/java/spring/intro/controller/UserController.java [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+        User user = userService.get(id);
+        return userMapper.parseToDto(user);
+    }
+
+    @GetMapping("/")
+    List<UserResponseDto> getAll() {
```
Comment (dmytro-hryhoruk) [inline src/main/java/spring/intro/controller/UserController.java:57]: and here

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**